src/Http/Request.php

```diff
@@ -351,9 +351,9 @@ protected static function parseHeadersFromGlobals() {
   /**
    * {@inheritdoc}
    */
-  public function getPath() {
+  public function getPath($strip = TRUE) {
```

Comment:
We use the `menu_router` table to detect the resource name based on a URL. In that table we have the URLs with like `api/v1.0/articles`, `api/v1/articles` and `api/articles`. If we remove `api/v1.0` from the beginning then we won't match any row in that table when querying.

We cannot derive the resource name by just exploring the URL, since a plugin may declare `menu_item = foo/bar/resource/big` where none of its parts is the name of the resource.